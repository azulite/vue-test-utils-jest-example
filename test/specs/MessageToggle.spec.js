import { mount } from '@vue/test-utils'
import MessageToggle from '@/components/MessageToggle.vue'
// import Message from '@/components/Message'

describe('MessageToggle.vue', () => {
  let wrapper
  // const spy = sinon.spy()

  // beforeEach(() => {
  //   wrapper = mount(MessageToggle, {
  //     // Beaware that props is overriden using `propsData`
  //     propsData: {
  //       msg: 'message'
  //     }
  //   })
  // })
  beforeEach(() => {
    wrapper = mount(MessageToggle, {
      // Beaware that props is overriden using `propsData`
      propsData: {
        msg: 'message'
      }
    })
  })

  // it('toggles msg passed to Message when button is clicked', () => {
  //   const button = wrapper.find('#toggle-message')
  //   button.trigger('click')
  //   spy.should.have.been.calledWith('yes')
  //   // expect(button.called).toBe(true)
  //   expect(wrapper.props().msg).toBe('message')
  //   button.trigger('click')
  //   // expect(button.called).toBe(true)
  //   expect(wrapper.props().msg).toBe('toggled message')
  // })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
