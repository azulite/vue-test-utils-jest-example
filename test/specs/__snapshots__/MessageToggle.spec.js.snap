// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageToggle.vue has the expected html structure 1`] = `
<div>
  <h1>
    default message
  </h1>
   
  <button
    id="toggle-message"
  >
    
  Change message
 
  </button>
</div>
`;
